server/src/test/java/org/elasticsearch/gateway/MetaDataStateFormatTests.java

```diff
@@ -382,14 +392,24 @@ public void testFailRandomlyAndReadAnyState() throws IOException {
             format.failRandomly();
             DummyState newState = new DummyState(randomRealisticUnicodeOfCodepointLengthBetween(1, 100), randomInt(), randomLong(),
                     randomDouble(), randomBoolean());
-            possibleStates.add(newState);
             try {
                 format.write(newState, paths);
-            } catch (Exception e) {
-                // since we're injecting failures at random it's ok if exception is thrown, it's also ok if exception is not thrown
+                //if write is successful, the only possible state we can read is the one that was just written
+                possibleStates.clear();
+                possibleStates.add(newState);
+            } catch (WriteStateException e) {
+                //if dirty flag is not set, read might return only old state
```

Comment:
`s/might/must/`. But I think I'd prefer to omit this comment and the next since they don't say anything that isn't in the Javadocs for `WriteStateException`.